File: examples/src/test/java/jdocs/classic/ClassicDocExample.java
/*
 * Copyright (C) 2020-2021 Lightbend Inc. <https://www.lightbend.com>
 */

package jdocs.classic;

import akka.actor.typed.ActorSystem;

import akka.projection.Projection;
import akka.projection.ProjectionBehavior;

// #import-adapter
import akka.actor.typed.javadsl.Adapter;

// #import-adapter

public interface ClassicDocExample {

  public static void illustrateSystem() {

    // #system
    akka.actor.ActorSystem system = akka.actor.ActorSystem.create("Example");
    ActorSystem<Void> typedSystem = Adapter.toTyped(system);
    // #system
  }

  public static void illustrateSpawn() {

    akka.actor.ActorSystem system = akka.actor.ActorSystem.create("Example");
    Projection<?> projection = null;

    // #spawn
    Adapter.spawn(system, ProjectionBehavior.create(projection), "theProjection");
    // #spawn
  }
}

File: examples/src/test/scala/docs/classic/ClassicDocExample.scala
/*
 * Copyright (C) 2020-2024 Lightbend Inc. <https://www.lightbend.com>
 */

package docs.classic

import akka.projection.Projection
import akka.projection.ProjectionBehavior

object ClassicDocExample {

  object IllustrateSystem {
    // #system
    import akka.actor.typed.scaladsl.adapter._

    private val system = akka.actor.ActorSystem("Example")
    private val typedSystem: akka.actor.typed.ActorSystem[_] = system.toTyped
    // #system

    typedSystem.terminate() // avoid unused warning
  }

  object IllustrateSpawn {
    private val system = akka.actor.ActorSystem("Example")
    private val projection: Projection[Any] = null

    //#spawn
    import akka.actor.typed.scaladsl.adapter._

    system.spawn(ProjectionBehavior(projection), "theProjection")
    //#spawn
  }
}
